java/trace/src/main/scala/org/typelevel/otel4s/java/trace/TracerImpl.scala

```diff
@@ -42,15 +45,17 @@ private[java] class TracerImpl[F[_]: Sync](
         None
     }
 
-  def spanBuilder(name: String): SpanBuilder[F] =
-    new SpanBuilderImpl[F](jTracer, name, scope)
+  def spanBuilder(name: String): SpanBuilder.Aux[F, Span[F]] =
+    new SpanBuilderImpl[F, Span[F]](jTracer, name, scope, simple)
 
-  def childScope(parent: SpanContext): Resource[F, Unit] =
-    scope.makeScope(JSpan.wrap(WrappedSpanContext.unwrap(parent)))
+  def childScope[A](parent: SpanContext)(fa: F[A]): F[A] =
+    scope
+      .makeScope(JSpan.wrap(WrappedSpanContext.unwrap(parent)))
+      .flatMap(_(fa))
 
-  def rootScope: Resource[F, Unit] =
-    scope.rootScope
+  def rootScope[A](fa: F[A]): F[A] =
+    scope.rootScope.flatMap(_(fa))
 
-  def noopScope: Resource[F, Unit] =
-    scope.noopScope
+  def noopScope[A](fa: F[A]): F[A] =
+    scope.noopScope(fa)
```

Comment:
Need to consider what these signature changes mean.  There's a failing test in `TracerSuite`, because we lost access to the span.